api/src/main/java/org/apache/cloudstack/api/command/admin/storage/SeedOfficialSystemVMTemplateCmd.java

```diff
@@ -0,0 +1,166 @@
+    private String hypervisor;
+
+    public Long getTemplateId() {
+        return templateId;
+    }
+
+    public void setTemplateId(Long templateId) {
+        this.templateId = templateId;
+    }
+
+    private Long templateId;
+
+    public String getInstallPath() {
+        return installPath;
+    }
+
+    public void setInstallPath(String installPath) {
+        this.installPath = installPath;
+    }
+
+    private String installPath;
+
+    public String getFileExtension() {
+        return fileExtension;
+    }
+
+    public void setFileExtension(String fileExtension) {
+        this.fileExtension = fileExtension;
+    }
+
+    private String fileExtension;
+
+
+    @Override
+    public void execute(){
+        // Get all the clusters for this zone
+        Pair<List<? extends Cluster>, Integer> clusters = _mgr.searchForClusters(this);
+        HashSet<String> hypervisors = new HashSet<>();
+        // we need to get the hypervisors from the clusters
+        for (Cluster cluster: clusters.first()){
+            hypervisors.add(cluster.getHypervisorType().toString());
+        }
+
+        // we need to get secondary storage for this zone.
+        HashSet<String> imageStores = _queryService.searchForImageStores(this);
+
+        String script = Script.findScript("scripts/storage/secondary/","cloud-install-sys-tmplt");
+
+        // mount locally
+        String mountPoint = "/tmp/nfsmount";
+        Script.runSimpleBashScript("mkdir " + mountPoint);
+
+        for (String imageStore: imageStores){
+            try {
+                URI uri = new URI(imageStore);
+                Script.runSimpleBashScript("sudo mount -t nfs " + uri.getHost() + ":" + uri.getPath() + " " + mountPoint);
+            } catch (URISyntaxException e) {
+                throw new CloudRuntimeException("Malformed URI " + imageStore);
+            }
+
+            // now download the template to the image store for every hypervisor if not specified
+            if (this.hypervisor != null && this.url != null){
+
+                if (this.hypervisor.equalsIgnoreCase("kvm") || this.hypervisor.equalsIgnoreCase("lxc")) {
+                    this.fileExtension = "qcow2";
+                } else if (this.hypervisor.equalsIgnoreCase("xenserver") || this.hypervisor.equalsIgnoreCase("hyperv")) {
+                    this.fileExtension = "vhd";
+                } else if (this.hypervisor.equalsIgnoreCase("vmware")) {
+                    this.fileExtension = "ova";
+                } else if (this.hypervisor.equalsIgnoreCase("ovm3")) {
+                    this.fileExtension = "raw";
+                }
+
+                this.templateId = _queryService.getSystemVMTemplateId(this);
+
+                HttpDirectTemplateDownloader httpDirectTemplateDownloader = new HttpDirectTemplateDownloader(url, templateId, mountPoint, null,null);
```

Comment:
Do not use this class, been checking it is duplicated and unused. You can invoke `wget URL -P mountPoint`

Reply:
I tried using wget. The wget script never ends and it never returns, the downloader works fine.

Reply:
Maybe it times out and fails silently